ddl-scripts/create-organization-tables.sql

```diff
@@ -0,0 +1,164 @@
+
+CREATE TABLE IF NOT EXISTS public.organization (
+    org_id uuid DEFAULT gen_random_uuid() PRIMARY KEY,
+    parent uuid,
+    name VARCHAR(255),
+    status organizationstatus,
+    members uuid[],
+    owners uuid[],
+    FOREIGN KEY (parent) REFERENCES organization(org_id)
+);
+CREATE TYPE public.organizationstatus AS ENUM ('ORGANIZATION_STATUS_DRAFT', 'ORGANIZATION_STATUS_ACTIVE', 'ORGANIZATION_STATUS_SUSPENDED', 'ORGANIZATION_STATUS_RELEASED', 'ORGANIZATION_STATUS_TERMINATED');
+CREATE TABLE IF NOT EXISTS public.info (
+    info_id uuid DEFAULT gen_random_uuid() PRIMARY KEY,
+    org_id uuid NOT NULL,
+    name VARCHAR(255) NOT NULL,
+    short_name VARCHAR(255),
+    is_private BOOLEAN,
+    url VARCHAR(255),
+    logo VARCHAR(255),
+    tenant_id uuid,
+--    address_id uuid,
+--    FOREIGN KEY (address_id) REFERENCES address(address_id),
+--    FOREIGN KEY (tenant_id) REFERENCES tenant(tenant_id),
+    FOREIGN KEY (org_id) REFERENCES organization(org_id),
+    UNIQUE(org_id)
+);
+CREATE TYPE public.postalcodetype AS ENUM ('caPostalCode', 'usPostalCode');
+CREATE TABLE IF NOT EXISTS public.address (
+    address_id uuid DEFAULT gen_random_uuid() PRIMARY KEY,
+    line1 VARCHAR(255) NOT NULL,
+    line2 VARCHAR(255),
+    city VARCHAR(255) NOT NULL,
+    state_province VARCHAR(255) NOT NULL,
+    country VARCHAR(255),
+    postal_code VARCHAR(255),
+    postal_code_type postalcodetype,
+    info_id uuid NOT NULL,
+    FOREIGN KEY (info_id) REFERENCES info(info_id),
+    UNIQUE(info_id)
+);
+CREATE TABLE IF NOT EXISTS public.metainfo (
+    metainfo_id uuid DEFAULT gen_random_uuid() PRIMARY KEY,
+    org_id uuid NOT NULL,
+    created_on TIMESTAMP NOT NULL,
+    created_by uuid NOT NULL,
+    last_updated TIMESTAMP,
+    last_updated_by uuid,
+    status organizationstatus,
+    children uuid[],
+    FOREIGN KEY (org_id) REFERENCES organization(org_id),
+    UNIQUE(org_id)
+);
+CREATE TABLE IF NOT EXISTS public.tenant (
+    tenant_id uuid DEFAULT gen_random_uuid() PRIMARY KEY,
+    org_id uuid NOT NULL,
+    FOREIGN KEY (org_id) REFERENCES organization(org_id),
+    UNIQUE(org_id)
+);
+CREATE TABLE IF NOT EXISTS public.member (
+    member_id uuid DEFAULT gen_random_uuid() PRIMARY KEY,
+    org_id uuid NOT NULL,
+--    meta_id uuid,
+--    FOREIGN KEY (meta_id) REFERENCES meta(meta_id),
+    FOREIGN KEY (org_id) REFERENCES organization(org_id)
+);
+CREATE TABLE IF NOT EXISTS public.owner (
+    owner_id uuid DEFAULT gen_random_uuid() PRIMARY KEY,
+    org_id uuid NOT NULL,
+    FOREIGN KEY (org_id) REFERENCES organization(org_id)
+);
+CREATE TABLE IF NOT EXISTS public.contact (
+    contact_id uuid DEFAULT gen_random_uuid() PRIMARY KEY,
+    primary_contacts uuid[],
+    billing_contacts uuid[],
+    distribution_contacts uuid[],
+    org_id uuid NOT NULL,
+    FOREIGN KEY (org_id) REFERENCES organization(org_id),
+    UNIQUE(org_id)
+);
+
+WITH insert_organization AS (
+INSERT INTO organization (org_id,
+		info_id,
```

Comment:
Here you are inserting `info_id` into `organization` table, but `info_id` was not declared as a field in `organization` table. The same applies to the `meta_id` field. Please ensure there are unit or integration tests to test that this is really working.

Reply:
I think I should remove this altogether as this was for the data model for relational db which is not used.  Cassandra is used for the projections.